R/convertToEGRET.R

```diff
@@ -88,32 +93,41 @@ convertToEGRETSample <- function(fitdat, meta, qconvert = 35.314667, dailydat =
   
   if(!is.null(dailydat)) {
     subDaily <- select(sample_df, dateTime) %>%
-      left_join(select(dailydat, dateTime, SE, yHat), by="dateTime") %>%
-      mutate(ConcHat = exp(yHat)*exp((SE^2)/2)) 
+      left_join(select(dailydat, dateTime, yHat, SE, ConcHat=ConcDay), by='dateTime')
+    # The sample-specific model predictions created by EGRET are actually 
+    # leave-one-out estimates where yHat, SE, and ConcHat are the predictions 
+    # from a model fit without the given observation. For the time being, we'll 
+    # stick to the simpler approach of using the same model for every row in the
+    # subDaily dataset. But LOOCV is the uncertainty estimation method for 
+    # loadInterp and half of the method for loadComp, so it would be appropriate
+    # for at least those models to use an approach like the EGRET approach here
+    # someday.
     
-    sample_df <- bind_cols(sample_df, subDaily)
+    sample_df <- left_join(sample_df, subDaily, by='dateTime')
   }
   
   return(sample_df)
 }
 
 #' Convert a loadflex metadata object into the EGRET INFO dataframe.
 #' 
-#' @param meta loadflex metadata object; it must include constituent,
-#' flow, dates, conc.units, site.id, and consti.name
-#' @param preds.type character specifying if the predictions being used are
-#' concentrations ("Conc") or fluxes ("Flux").
-#' 
+#' @param meta loadflex metadata object; it must include site.name, consti.name,
+#'   site.id, constituent, and the relevant type of units (conc.units or
+#'   load.units, depending on preds.type)
+#' @param preds.type character specifying if the predictions being used are 
+#'   concentrations ("Conc") or fluxes ("Flux").
+#'   
 convertToEGRETInfo <- function(meta, preds.type = 'Conc') {
   if(is.null(meta)) {
     stop("metadata is required to create an EGRET eList")
   }
+  match.arg(preds.type)
```

Comment:
It's a placeholder on the way to eliminating `preds.type` altogether. It checks the value of `preds.type` against the possible values listed in the function declaration (in this case, the only option is 'Conc'). If the given option is one of those possible values, it does nothing. If the given value is not one of those possible values, it throws an error. So I'm just putting in code to prevent a user from specifying anything other than the default, 'Conc', because #170.